xml_loader/shift_generation.py

```diff
@@ -0,0 +1,261 @@
+from gurobipy import *
+from pathlib import Path
+from xml_loader.xml_loader import *
+import xml.etree.ElementTree as ET
+
+
+def get_time_steps(root):
+    demands = get_demand_definitions(root)
+    time_step_length = 100
+    for demand in demands:
+        for i in range(len(demand.end)):
+            if(demand.end[i] - int(demand.end[i]) > 0):
+                if(demand.end[i] - int(demand.end[i]) == 0.5 and time_step_length != 0.5):
+                    time_step_length = 0.5
+                elif(demand.end[i] - int(demand.end[i]) in [0.25,0.75] and time_step_length != 0.25):
+                    time_step_length = 0.25
+                elif(demand.end[i] - int(demand.end[i]) < 0.25):
+                    time_step_length = 100/60
```

Comment:
Hva skjer her?

Reply:
Sjekker om man bruker kvarter, minutter, halvtimer eller timer som timestep

Reply:
Tenkte spesifikt på verdien 100/60. Hva representerer den? Burde det ikke vært 1/60 dersom det skal være minutt?

Reply:
1 minutt når man bruker 100 som time vil være 1.6666666. Bruker 100/60 ettersom dette er mer nøyaktig.

Reply:
No wait. Du har helt rett

Reply:
Oppdater nå! Endret til 1/60. 

Har også fikset bugs som ga time_step 0.5 på rproblem4, og ikke 0.25 som den burde. Fungerer nå, og testet på de øvrige probleminstansen as well.

Merk det jeg har kommentert ut, tror det er overflødig kode. Bet meg også merke i de to elif-ene som er kommentert på. Tror ikke disse  gir mening, se TODO i koden for  hvorfor.